nest/src/token/pUSD.sol

```diff
@@ -295,51 +278,70 @@ contract pUSD is
     }
 
     /**
-     * @notice Burn shares and withdraw corresponding assets
+     * @notice Request to redeem shares for assets through the atomic queue
      * @param shares Amount of shares to burn
      * @param receiver Address that will receive the assets
-     * @param controller Address that currently controls the shares
-     * @return assets Amount of assets withdrawn
+     * @param controller Address that will control the redemption
+     * @param price Price in terms of underlying asset (e.g., USDC) per share
+     * @param deadline Deadline for the atomic request
      */
-    function redeem(
+    function requestRedeem(
         uint256 shares,
         address receiver,
         address controller,
         uint256 price,
         uint64 deadline
```

Comment:
Doesn't match the `IComponentToken` interface - there shouldn't be a price or a deadline

Reply:
but boringvault needs price and deadline for AtomicRequest in requestRedeem function? let me think about it.